symLogging.py

```diff
@@ -1,29 +1,58 @@
 import sys
 import threading
 import time
+import logging
+import os.path
+import tempfile
+from logging.handlers import RotatingFileHandler
 
-gEnableTracing = False
+gLog = None
 
 # Whether to pause and launch debugger at CheckDebug breakpoints
 gDebug = False
 
-def SetTracingEnabled(isEnabled):
-  global gEnableTracing
-  gEnableTracing = isEnabled
+def SetLoggingOptions(logOptions):
+  global gLog
 
-def LogTrace(string):
-  global gEnableTracing
-  if gEnableTracing:
+  fmt = logging.Formatter('%(asctime)s\t%(levelname)s\t%(message)s')
+  streamHandler = logging.StreamHandler()
+  streamHandler.setFormatter(fmt)
+  gLog = logging.getLogger("Snappy")
+  gLog.addHandler(streamHandler)
+
+  filepath = logOptions.get('logPath', tempfile.gettempdir())
+
+  try:
+    rotatingHandler = RotatingFileHandler(
+                filename=os.path.join(filepath, 'snappy.log'),
+                mode='a',
+                maxBytes=int(logOptions.get("maxFileSize", 1024*1024)),
+                backupCount=int(logOptions.get("maxFiles", 10)))
+
+    rotatingHandler.setFormatter(fmt)
+    gLog.addHandler(rotatingHandler)
+  except IOError as ex:
+    gLog.error("Could not configure file logger: %s", ex)
+    gLog.error("Check the log path '%s'. Logging on screen only.", filepath)
+
+  logLevel = getattr(logging, logOptions.get("logLevel", "INFO"))
+  gLog.setLevel(logLevel)
+
+def doLog(dbgLevel, string):
     threadName = threading.currentThread().getName().ljust(12)
-    print >> sys.stdout, time.asctime() + " " + threadName + " TRACE " + string
+    gLog.log(dbgLevel, "%s\t%s", threadName, string)
+
+def LogTrace(string):
```

Comment:
nit: might as well rename this to LogDebug, we don't have enableTracing in the config file anymore